src/usr/local/www/services_unbound_advanced.php

```diff
@@ -335,10 +335,12 @@
 	'edns_buffer_size',
 	'EDNS Buffer Size',
 	$pconfig['edns_buffer_size'],
-	array_combine(array("512", "1480", "4096"), array("512", "1480", "4096"))
-))->setHelp('Number of bytes size to advertise as the EDNS reassembly buffer size. This is the value that is used in UDP datagrams sent to peers. ' .
-			'RFC recommendation is 4096 (which is the default). If fragmentation reassemble problems occur, usually seen as timeouts, then a value of 1480 should help. ' .
-			'The 512 value bypasses most MTU path problems, but it can generate an excessive amount of TCP fallback.');
+	array_combine(array("auto", "512", "1232", "1432", "1480", "4096"), array("Auto", "512", "1232", "1432", "1480", "4096"))
```

Comment:
The array of choices would be better as a global set at the start of the page, with the choices as keys and some explanation about why users might choose each one in the value.

Something like this:

```
$unbound_edns_sizes = array(
	"auto" => "Automatic value based on active interface MTUs",
	"512"  => "512: IPv4 Minimum",
	"1220" => "1220: NSD IPv6 EDNS Minimum",
	"1232" => "1232: IPv6 Minimum",
	"1432" => "1432: 1500 Byte MTU",
	"1480" => "1480: NSD IPv4 EDNS Minimum",
	"4096" => "4096: Unbound Default"
);
```

And then you can use that array when doing the tests (e.g. `array_key_exists( ... )` in the validity check, and using the array directly here in the field.

Then you can simplify the description text below as well since you don't have to explain them all.

Reply:
ok, done
I see no reason to simplify the description text - it does not explain all available sizes, but provides useful info for fragmentation and MTU issues.